lib/DB/Schema/ResultSet/Course.pm

```diff
@@ -262,36 +335,202 @@ This gets the Course Settings for a course
+=item * C<info>, hashref containing info about the course
+
+=item * C<merged>, a boolean on whether the course setting is merged with its corresponding
+global setting.
 
 =item * C<$as_result_set>, a boolean if the return is to be a result_set
 
 =back
 
 =head3 output
 
-An array of courses as a C<DBIx::Class::ResultSet::Course> object
+An array of course settings as a C<DBIx::Class::ResultSet::CourseSetting> object
 if C<$as_result_set> is true.  Otherwise an array of hash_ref.
 
 =cut
 
 sub getCourseSettings ($self, %args) {
-	my $course = $self->getCourse(info => $args{info}, as_result_set => 1);
+	my $course           = $self->getCourse(info => $args{info}, as_result_set => 1);
+	my @settings_from_db = $course->course_settings;
+
+	return \@settings_from_db if $args{as_result_set};
+	my @settings_to_return = ($args{merged})
+		? map {
+			{ $_->get_inflated_columns, $_->global_setting->get_inflated_columns };
+		} @settings_from_db
+		: map {
+			{ $_->get_inflated_columns };
+		} @settings_from_db;
+
+	for my $setting (@settings_to_return) {
+		# value and default_value are decoded from JSON as a hash.  Return to a value.
+		for my $key (qw/default_value value/) {
+			$setting->{$key} = $setting->{$key}->{value} if defined($setting->{$key});
+		}
+	}
+	return \@settings_to_return;
+}
+
+=pod
+
+=head2 getCourseSetting
+
+This gets a single course setting.
+
+=head3 input
+
+=over
+
+=item * C<info> which is a hash of either a C<setting_id> or C<setting_name> with information
+on the setting.
+
+=item * C<merged>, a boolean on whether the course setting is merged with its corresponding
+global setting.
+
+=item * C<$as_result_set>, a boolean if the return is to be a result_set
+
+=back
+
+=head3 output
+
+A single course setting as either a hashref or a C<DBIx::Class::ResultSet::CourseSetting> object.
+
+=cut
+
+sub getCourseSetting ($self, %args) {
+
+	my $global_setting = $self->getGlobalSetting(info => $args{info}, as_result_set => 1);
+	DB::Exception::SettingNotFound->throw(
+		message => "The setting with name: '" . $args{info}->{setting_name} . "' is not a defined info.")
+		unless defined($global_setting);
+
+	my $course  = $self->getCourse(info => getCourseInfo($args{info}), as_result_set => 1);
+	my $setting = $course->course_settings->find({ setting_id => $global_setting->setting_id });
+
+	return $setting if $args{as_result_set};
+	my $setting_to_return =
+		$args{merged}
+		? { $setting->get_inflated_columns, $setting->global_setting->get_inflated_columns }
+		: { $setting->get_inflated_columns };
 
-	my $course_settings  = getDefaultCourseValues();
-	my $settings_from_db = { $course->course_settings->get_inflated_columns };
-	return mergeCourseSettings($course_settings, $settings_from_db);
+	# value and default_value are decoded from JSON as a hash.  Return to a value.
+	for my $key (qw/default_value value/) {
+		$setting_to_return->{$key} = $setting_to_return->{$key}->{value} if defined($setting_to_return->{$key});
+	}
```

Comment:
It seems that we're doing this across multiple calls... why not create a util function that will do an in-place edit on a hash?
```perl
sub hash2value ($hash) {
    for (qw/default_value value/) {
        $hash->{$_} = $hash->{$_}{value} if defined($hash->{$_}{value});
    }
}
```

Reply:
This sounds good.  I also commented above on if this would be helpful to make an inflate/deflate function that does this.